src/main/java/org/gradle/profiler/mutations/support/ProjectCombinationsSupport.java

```diff
@@ -0,0 +1,44 @@
+package org.gradle.profiler.mutations.support;
+
+import com.google.common.collect.Sets;
+import com.google.common.math.IntMath;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.gradle.profiler.mutations.ApplyProjectDependencyChangeMutatorConfigurator.APPLIED_PROJECTS_COUNT_KEY;
+
+public class ProjectCombinationsSupport {
```

Comment:
Why not have these methods on `ProjectCombinations`?

Reply:
I really like "data only classes". So classes that don't have any logic in there (except maybe trivial creator methods), since that makes other code easier to test also soon that data class can get bloated with similar static methods.

This particular logic is already a bit complex so I decided to move out of it. Since this is a creater logic, I can move it into the class.

Reply:
removed support class